python/mxnet/symbol/symbol.py

```diff
@@ -1078,11 +1078,22 @@ def _infer_shape_impl(self, partial, *args, **kwargs):
             keys = c_array(ctypes.c_char_p, [])
             for i, s in enumerate(args):
                 if s is not None:
-                    if not isinstance(s, tuple):
-                        raise TypeError("Arguments need to be shapes (tuple), "
-                                        "but argument %d is %s." % (i, type(s)))
-                    sdata.extend(s)
-                indptr.append(len(sdata))
+                    if isinstance(s, dict):
+                        str_keys = []
+                        for k, v in s.items():
+                            if not isinstance(v, tuple):
+                                raise TypeError("Arguments need to be shapes (tuple), "
```

Comment:
Thanks for this fix.

1. Same checks in 2 places. Please refactor.
2. Validation on k?
3. indptr.append(..) can be moved of this logic.